src/EditorFeatures/Core/FindUsages/AbstractFindUsagesService.cs

```diff
@@ -95,5 +108,71 @@ internal abstract partial class AbstractFindUsagesService : IFindUsagesService
 
             return progressAdapter;
         }
+
+        private async Task<bool> TryFindLiteralReferencesAsync(
+            Document document, int position, IFindUsagesContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var token = await syntaxTree.GetTouchingTokenAsync(
+                position,
+                t => syntaxFacts.IsNumericLiteral(t) ||
+                     syntaxFacts.IsCharacterLiteral(t) ||
+                     syntaxFacts.IsStringLiteral(t),
+                cancellationToken).ConfigureAwait(false);
+
+            if (token.RawKind == 0)
+            {
+                return false;
+            }
+
+            // Searching for decimals not supported currently.
+            var tokenValue = token.Value;
+            if (tokenValue == null || tokenValue is decimal)
+            {
+                return false;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var symbol = semanticModel.GetSymbolInfo(token.Parent).Symbol ?? semanticModel.GetDeclaredSymbol(token.Parent);
+            if (symbol is ILabelSymbol)
+            {
+                return false;
+            }
+
+            // Use the literal to make the title.  Trim literal if it's too long.
+            var title = syntaxFacts.ConvertToSingleLine(token.Parent).ToString();
+            if (title.Length >= 10)
+            {
+                title = title.Substring(0, 10) + "...";
+            }
+
+            var searchTitle = string.Format(EditorFeaturesResources._0_references, title);
+            context.SetSearchTitle(searchTitle);
+
+            var solution = document.Project.Solution;
+
+            // There will only be one 'definition' that all matching literal reference.
+            // So just create it now and report to the context what it is.
+            var definition = DefinitionItem.CreateNonNavigableItem(
+                ImmutableArray.Create(TextTags.StringLiteral),
+                ImmutableArray.Create(new TaggedText(TextTags.Text, searchTitle)));
+
+            await context.OnDefinitionFoundAsync(definition).ConfigureAwait(false);
+
+            var progressAdapter = new FindLiteralsProgressAdapter(context, definition);
+
+            // Now call into the underlying FAR engine to find reference.  The FAR
+            // engine will push results into the 'progress' instance passed into it.
+            // We'll take those results, massage them, and forward them along to the 
+            // FindUsagesContext instance we were given.
+            await SymbolFinder.FindLiteralReferencesAsync(
```

Comment:
massaging is done.  see the first comment i made about the two different FAR systems.  The 'ProgressAdapter' massages like so:

```c#
            public Task OnReferenceFoundAsync(Document document, TextSpan span)
                => _context.OnReferenceFoundAsync(new SourceReferenceItem(
                    _definition, new DocumentSpan(document, span)));
```